src/Microsoft.ML.StandardLearners/NearestNeighbor/NearestNeighbor.cs

```diff
@@ -0,0 +1,247 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.ML.Core.Data;
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.CommandLine;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Runtime.EntryPoints;
+using Microsoft.ML.Runtime.Internal.Internallearn;
+using Microsoft.ML.Runtime.Internal.Utilities;
+using Microsoft.ML.Runtime.Model;
+using Microsoft.ML.Runtime.Numeric;
+using Microsoft.ML.Runtime.Training;
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.IO;
+
+namespace Microsoft.ML.Trainers
+{
+    public abstract class NearestNeighborBase<TArgs, TTransformer, TModel> : TrainerEstimatorBase<TTransformer, TModel>
+      where TTransformer : ISingleFeaturePredictionTransformer<TModel>
+      where TModel : IPredictor
+      where TArgs : NearestNeighborBase<TArgs, TTransformer, TModel>.ArgumentsBase, new()
+    {
+        public abstract class ArgumentsBase : LearnerInputBaseWithLabel
+        {
+            [Argument(ArgumentType.AtMostOnce, HelpText = "Number of neighbors to use by default to determine result.", SortOrder = 50)]
+            [TGUI(SuggestedSweeps = "0,5,10,20,40")]
+            [TlcModule.SweepableDiscreteParam("K", new object[] { 0, 5, 10, 20, 40 })]
```

Comment:
Is `0` an acceptable number of nearest neighbors to look at?

Assuming we're starting at `K=1`, I may recommend a log scale sweepable long param. 

```suggestion
            [TGUI(Description = "Number of neighbors to use by default to determine result", SuggestedSweeps = "1-64;log;inc:2")]
            [TlcModule.SweepableLongParamAttribute("K", 2, 64, isLogScale: true, stepSize: 2)]
```